File: beer-store-service-jee/src/main/java/beer/store/model/Beer.java
package beer.store.model;

import java.util.Date;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.Id;
import javax.persistence.Table;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;
import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlRootElement;

@Entity
@XmlRootElement
@Table(name = "beer")
@XmlAccessorType(XmlAccessType.FIELD)
public class Beer {

	@Id
	@XmlElement(name = "idbeer")
	@Column(name = "idbeer")
	private long idBeer;

	@Temporal(TemporalType.TIMESTAMP)
	@Column(name = "creationdatebeer")
	private Date creationDateBeer;

	@Column(name = "titlebeer")
	private String titleBeer;

	@Column(name = "descriptionbeer")
	private String descriptionBeer;

	@Column(name = "idmedia")
	private Long idMedia;

	public long getIdBeer() {
		return idBeer;
	}

	public void setIdBeer(long idBeer) {
		this.idBeer = idBeer;
	}

	public Date getCreationDateBeer() {
		return creationDateBeer;
	}

	public void setCreationDateBeer(Date creationDateBeer) {
		this.creationDateBeer = creationDateBeer;
	}

	public String getTitleBeer() {
		return titleBeer;
	}

	public void setTitleBeer(String titleBeer) {
		this.titleBeer = titleBeer;
	}

	public String getDescriptionBeer() {
		return descriptionBeer;
	}

	public void setDescriptionBeer(String descriptionBeer) {
		this.descriptionBeer = descriptionBeer;
	}

	public Long getIdMedia() {
		return idMedia;
	}

	public void setIdMedia(Long idMedia) {
		this.idMedia = idMedia;
	}

}

File: beer-store-client-qt/src/model/beer.cpp
#include "beer.h"

Beer::Beer(QObject *parent) : QObject(parent) {}

Beer::Beer(const Beer &beer) : QObject(beer.parent()) {
  this->idbeer = beer.idbeer;
  this->creationdatebeer = beer.creationdatebeer;
  this->titlebeer = beer.titlebeer;
  this->descriptionbeer = beer.descriptionbeer;
  this->idmedia = beer.idmedia;
}

Beer *Beer::fromJson(QJsonValue val) {
  Beer *b = new Beer();
  QJsonObject o = val.toObject();
  b->idbeer = o.value("idbeer").toInt();
  b->creationdatebeer =
      QDateTime::fromString(o.value("creationdatebeer").toString());
  b->titlebeer = o.value("titlebeer").toString();
  b->descriptionbeer = o.value("descriptionbeer").toString();
  b->idmedia = o.value("idmedia").toInt();
  return b;
}

Beer &Beer::operator=(Beer &other) {
  Beer *b = new Beer(other);
  return *b;
}

int Beer::getIdbeer() { return idbeer; }

void Beer::setIdbeer(int idbeer) { this->idbeer = idbeer; emit idbeerChanged(); }

QDateTime Beer::getCreationdatebeer() { return creationdatebeer; }

void Beer::setCreationdatebeer(QDateTime creationdatebeer) {
  this->creationdatebeer = creationdatebeer;
  emit creationdatebeerChanged();
}

QString Beer::getTitlebeer() { return titlebeer; }

void Beer::setTitlebeer(QString titlebeer) {
  this->titlebeer = titlebeer;
  emit titlebeerChanged();
}

QString Beer::getDescriptionbeer() { return descriptionbeer; }

void Beer::setDescriptionBeer(QString descriptionbeer) {
  this->descriptionbeer = descriptionbeer;
  emit descriptionbeerChanged();
}

int Beer::getIdmedia() { return idmedia; }

void Beer::setIdmedia(int idmedia) {
  this->idmedia = idmedia;
  emit idmediaChanged();
}
